crypto/fipsmodule/bcm.c

```diff
@@ -228,7 +228,7 @@ static void BORINGSSL_bcm_power_on_self_test(void) __attribute__ ((constructor))
 #endif
 
 static void BORINGSSL_bcm_power_on_self_test(void) {
-#if !defined(OPENSSL_NO_ASM)
+#if !defined(OPENSSL_NO_ASM) && !defined(OPENSSL_PPC32BE)
```

Comment:
Would it be better to set `OPENSSL_NO_ASM` define in the CMakeLists.txt when you detect big-endian / if the processor is ppc64 in the CMakeLists.txt?

Reply:
IMHO, this should define `OPENSSL_ASM_INCOMPATIBLE` (which is later evaluated to `OPENSSL_NO_ASM`) in `include/openssl/base.h` when defining platform-specific variables, since there is a precedent for that there.

Doing this in `CMakeLists.txt` splits logic between build system and source code, which is neither ideal nor necessary in this case.